pkg/resolution/resource/name.go

```diff
@@ -23,24 +23,58 @@ import (
 	"sort"
 
 	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"github.com/tektoncd/pipeline/pkg/apis/resolution/v1beta1"
+	"knative.dev/pkg/kmeta"
 )
 
+// GenerateDeterministicName makes a best-effort attempt to create a
+// unique but reproducible name for use in a Request. The returned value
+// will have the format {prefix}-{hash} where {prefix} is
+// given and {hash} is nameHasher(base) + nameHasher(param1) +
+// nameHasher(param2) + ...
+func GenerateDeterministicName(prefix, base string, params v1.Params) (string, error) {
```

Comment:
Does this algorithm provide a fixed-length name (apart from the prefix)?
Can we ensure that names will fit k8s requirements for names?

Reply:
Good point. I don't think it asserts anything. Let me fix that.

Reply:
fixed!